sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -168,4 +168,89 @@ private static boolean isValid(String name) {
   private static boolean isValidAndNotEmpty(AttributeKey<?> name) {
     return !name.getKey().isEmpty() && isValid(name.getKey());
   }
+
+  public static Resource.Builder builder() {
+    return new Resource.Builder();
+  }
+
+  public Builder toBuilder() {
```

Comment:
If we add this to incubator first, we would add `ResourceBuilder.from(Resource` and not add these methods yet. This class seems genuinely useful though and maybe we can go with it without incubator. @jkwatson what do you think?